egg/core/callbacks.py

```diff
@@ -207,3 +217,115 @@ def on_test_end(self, loss: float, logs: Interaction, epoch: int):
     def on_epoch_end(self, loss: float, logs: Interaction, epoch: int):
         if epoch in self.train_epochs:
             self.dump_interactions(logs, "train", epoch, self.folder_path)
+
+
+class ProgressBarLogger(Callback):
+    class CompletedColumn(ProgressColumn):
+        def render(self, task):
+            """Calculate common unit for completed and total."""
+            download_status = f"{int(task.completed)}/{int(task.total)} btc"
+            return Text(download_status, style="progress.download")
+
+    class TransferSpeedColumn(ProgressColumn):
+        """Renders human readable transfer speed."""
+
+        def render(self, task):
+            """Show data transfer speed."""
+            speed = task.speed
+            if speed is None:
+                return Text("?", style="progress.data.speed")
+            speed = f"{speed:,.{2}f}"
+            return Text(f"{speed} btc/s", style="progress.data.speed")
+
+    def __init__(
+        self,
+        n_epochs,
+        train_data_len=None,
+        eval_data_len=None,
+    ):
+        self.n_epochs = n_epochs
+        self.cur_epoch = 1
+        self.train_data_len = train_data_len
+        self.eval_data_len = eval_data_len
+        self.progress = Progress(
+            TextColumn(
+                "[bold]Epoch {task.fields[cur_epoch]}/{task.fields[n_epochs]} | [blue]{task.fields[mode]}",
+                justify="right",
+            ),
+            BarColumn(bar_width=None),
+            "[progress.percentage]{task.percentage:>3.1f}%",
+            "•",
+            self.CompletedColumn(),
+            "•",
+            self.TransferSpeedColumn(),
+            "•",
+            TimeRemainingColumn(),
+        )
+
+        self.progress.start()
+        self.train_p = self.progress.add_task(
+            description="",
+            mode="Train",
+            cur_epoch=self.cur_epoch,
+            n_epochs=self.n_epochs,
+            start=False,
+            visible=False,
+            total=self.train_data_len,
```

Comment:
Does it work no problem when None data_len is passed?

Reply:
yes

Reply:
My bad you were right.
Fixed in 6eedce9